deepchem/trans/transformers.py

```diff
@@ -2469,3 +2469,120 @@ def median_filter(self, size):
     image = Image.fromarray(self.Image)
     image = image.filter(ImageFilter.MedianFilter(size=size))
     return np.array(image)
+
+
+class RxnSplitTransformer(Transformer):
+  """Splits the reaction SMILES input into the source and target strings
+  required for machine translation tasks.
+
+  The input is expected to be in the form reactant>reagent>product. The source
+  string would be reactants>reagents and the target string would be the products.
+
+  The transformer can also separate the reagents from the reactants for a mixed
+  training mode. This can be toggled (default True) by setting the value of
+  sep_reagent while calling the transformer.
+
+  Examples
+  --------
+  >>> # When mixed training is toggled.
+  >>> import numpy as np
+  >>> from deepchem.trans.transformers import RxnSplitTransformer
+  >>> reactions = np.array(
+    [
+        "CC(C)C[Mg+].CON(C)C(=O)c1ccc(O)nc1>C1CCOC1.[Cl-]>CC(C)CC(=O)c1ccc(O)nc1",
```

Comment:
Can you simplify this to have only 2 reactions instead of 3? Will make it easier to read

Reply:
Yup, I will do that. Only 2 are needed to demonstrate the mixed training functionality.